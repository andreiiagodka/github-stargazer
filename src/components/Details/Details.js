import React, { Component } from "react";

class Details extends Component {
  render() {
    return (
      <h1>Details page</h1>
    )
  }
}

export default Details