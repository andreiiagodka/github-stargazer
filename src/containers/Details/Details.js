import React, { Component, Fragment } from "react";
import {
  Row,
  Col,
  Card,
  Button,
  Container
} from "react-bootstrap";
import { connect } from 'react-redux';
import { Redirect, withRouter } from 'react-router';

import * as actions from '../../store/actions/actions';
import { deleteRepository } from '../../firebase/firebase'

import Header from '../../components/UI/Header/Header';
import Spinner from '../../components/UI/Spinner/Spinner';
import ActionButton from '../../components/Details/ActionButton/ActionButton';
import CollectionList from '../../components/Details/CollectionList/CollectionList';

class Show extends Component {
  componentDidMount() {
    this.props.getRepository(this.props.match.params.id)
  }

  handleDelete = (id) => {
    deleteRepository(id)
    this.props.history.push('/')
  }

  render() {
    let content = <Spinner />
    if (!this.props.loading) {
      const actionButton = <ActionButton handleDelete={() => this.handleDelete(this.props.repository.id)} />

      content = (
        <Fragment>
          <Card.Header>
            <Header title={this.props.repository.full_name} actionButton={actionButton} />
          </Card.Header>
          <Card.Body>
            <Container>
              <Row>
                <h3>Stats</h3>
              </Row>
              <Row>
                <ul style={{ listStyleType: "none" }}>
                  <li>Stars: {this.props.repository.stargazers_count}</li>
                  <li>Watchers: {this.props.repository.watchers_count}</li>
                  <li>Forks: {this.props.repository.forks_count}</li>
                </ul>
              </Row>
            </Container>
            <CollectionList 
              title='Languages'
              collection={this.props.repository.languages} />
          </Card.Body>
        </Fragment>
      )
    }

    return content
  }
}

const mapStateToProps = state => {
  return {
    repository: state.getRepository.repository,
    loading: state.getRepository.loading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getRepository: (id) => dispatch(actions.getRepository(id))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Show))