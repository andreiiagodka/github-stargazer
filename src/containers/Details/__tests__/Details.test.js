import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { MemoryRouter } from 'react-router'
import Details from '../Details'

describe('<Details /> container', () => {
  let store

  const mockStore = configureStore([])
  const fetchRepository = {
    repository: {},
    loading: false,
    error: false,
    fetchRepository: jest.fn()
  }

  it('renders <Spinner /> container', () => {
    const fetchRepositoryProps = {
      ...fetchRepository,
      loading: true
    }
    store = mockStore({ fetchRepository: fetchRepositoryProps })
    store.dispatch = jest.fn()

    const tree = renderer.create(
      <Provider store={store} >
        <MemoryRouter>
          <Details />
        </MemoryRouter>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders <Error /> container', () => {
    const fetchRepositoryProps = {
      ...fetchRepository,
      error: true
    }
    store = mockStore({ fetchRepository: fetchRepositoryProps })
    store.dispatch = jest.fn()

    const tree = renderer.create(
      <Provider store={store} >
        <MemoryRouter>
          <Details />
        </MemoryRouter>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders container correctly', () => {
    const fetchRepositoryProps = {
      ...fetchRepository,
      repository: {
        id: '1',
        full_name: 'Full name',
        stargazers_count: 1,
        watchers_count: 2,
        forks_count: 3,
        languages: { 'Ruby': 4 }
      }
    }
    store = mockStore({ fetchRepository: fetchRepositoryProps })
    store.dispatch = jest.fn()

    const tree = renderer.create(
      <Provider store={store} >
        <MemoryRouter>
          <Details />
        </MemoryRouter>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})