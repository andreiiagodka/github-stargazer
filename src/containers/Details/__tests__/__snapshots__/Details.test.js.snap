// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> container renders <Error /> container 1`] = `
<div
  className="fade text-center alert alert-danger show"
  role="alert"
>
  Something went wrong!
</div>
`;

exports[`<Details /> container renders <Spinner /> container 1`] = `
<div
  className="Loader"
>
  Loading ...
</div>
`;

exports[`<Details /> container renders container correctly 1`] = `
Array [
  <div
    className="card-header"
  >
    <div
      className="row"
    >
      <div
        className="col-md-2"
      />
      <div
        className="text-center col-md-8"
      >
        <h5>
          Full name
        </h5>
      </div>
      <div
        className="text-right col-md-2"
      >
        <button
          className="btn btn-light"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-trash fa-w-14 "
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="card-body"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <h3>
          Stats
        </h3>
      </div>
      <div
        className="row"
      >
        <ul
          style={
            Object {
              "listStyleType": "none",
            }
          }
        >
          <li>
            Stars: 
            1
          </li>
          <li>
            Watchers: 
            2
          </li>
          <li>
            Forks: 
            3
          </li>
        </ul>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <h3>
          Languages
        </h3>
      </div>
      <div
        className="row"
      >
        <ul
          style={
            Object {
              "listStyleType": "none",
            }
          }
        >
          <li>
            Ruby
            : 
            4
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;
